src/utils.ts

```diff
@@ -33,6 +33,7 @@ export function documentTemplates<D extends Document>(
   const templatesFromAttachments = (document.attachments || []).map((attachment, index) => ({
     id: `attachment-${index}`,
     label: attachment.filename || "Unknown filename",
+    type: attachment.type || "Unknown filetype",
     template: attachmentToComponent(attachment, document)
```

Comment:
I feel this line is wrong and makes things confusing in the code but for simplicity I would suggest the following:
- introduction of a new type calls TemplateWithTypes in src/types.ts
```
export interface TemplateWithTypes<D extends Document> extends Template<D> {
  type: string;
}
```
- documentTemplates must return TemplateWithTypes (you should see an error and need to give a default value to selectedTemplate
- updateTemplates from `frame.actions.ts` must expect a type property along the id and label


If you want to hear more about what's wrong with the `template` property here and want to help to fix, let me know